src/rust/kafka/src/lib.rs

```diff
@@ -578,12 +578,6 @@ impl RetryProcessor {
                     result.map(|_| ())
                 }
             })
-            .then(|result| async {
-                if let Ok(_) = result {
-                    Ok(self.consumer.commit()?)
-                } else {
-                    result
-                }
-            })
+            .then(move |result| async { result.and_then(|_| Ok(self.consumer.commit()?)) })
```

Comment:
I'd imagine you could remove the Ok() and the ?

Reply:
but if not, whatever

Reply:
maybe the ? is doing some conversion for us

Reply:
Yes, the `?` is converting the `CommitError` for us